providers/redhat/schema/convert.go

```diff
@@ -0,0 +1,220 @@
+	}
+
+	return &nvd.CVEJSON40References{ReferenceData: referenceData}
+}
+
+func (cve *CVE) newImpact() (*nvd.NVDCVEFeedJSON10DefImpact, error) {
+	if cve.CVSS == nil && cve.CVSS3 == nil {
+		return nil, fmt.Errorf("cvss v2 nor cvss v3 is set in the cve")
+	}
+
+	impact := nvd.NVDCVEFeedJSON10DefImpact{}
+
+	if cve.CVSS != nil {
+		score, err := strconv.ParseFloat(cve.CVSS.BaseScore, 64)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse cvss v2 base score: %v", err)
+		}
+		impact.BaseMetricV2 = &nvd.NVDCVEFeedJSON10DefImpactBaseMetricV2{
+			CVSSV2: &nvd.CVSSV20{
+				BaseScore:    score,
+				VectorString: cve.CVSS.Vector,
+			},
+		}
+	}
+
+	if cve.CVSS3 != nil {
+		score, err := strconv.ParseFloat(cve.CVSS3.BaseScore, 64)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse cvss v3 base score: %v", err)
+		}
+		impact.BaseMetricV3 = &nvd.NVDCVEFeedJSON10DefImpactBaseMetricV3{
+			CVSSV3: &nvd.CVSSV30{
+				BaseScore:    score,
+				VectorString: cve.CVSS3.Vector,
+			},
+		}
+	}
+
+	return &impact, nil
+}
+
+// CPEs configuration, AKA the tricky part
+
+func (cve *CVE) newConfigurations() (*nvd.NVDCVEFeedJSON10DefConfigurations, error) {
+	nodes := make([]*nvd.NVDCVEFeedJSON10DefNode, len(cve.AffectedRelease)+len(cve.PackageState))
+
+	var err error
+
+	for i, ar := range cve.AffectedRelease {
+		if nodes[i], err = ar.createNode(); err != nil {
+			return nil, fmt.Errorf("can't create node for affected release %d: %v", i, err)
+		}
+	}
+
+	offset := len(cve.AffectedRelease)
+	for i, ps := range cve.PackageState {
+		if nodes[i+offset], err = ps.createNode(); err != nil {
+			return nil, fmt.Errorf("can't create node for package state %d: %v", i, err)
+		}
+	}
+
+	conf := nvd.NVDCVEFeedJSON10DefConfigurations{
+		CVEDataVersion: cveVersion,
+		Nodes:          nodes,
+	}
+
+	return &conf, nil
+}
+
+func (ar *AffectedRelease) createNode() (*nvd.NVDCVEFeedJSON10DefNode, error) {
+	node := nvd.NVDCVEFeedJSON10DefNode{
+		Operator: "AND",
+		CPEMatch: []*nvd.NVDCVEFeedJSON10DefCPEMatch{
+			{
+				Cpe22Uri:   ar.CPE,
+				Vulnerable: true,
+			},
+		},
+	}
+
+	if ar.Package != "" {
+		pkgAttrs, err := package2wfn(ar.Package)
+		if err != nil {
+			return nil, fmt.Errorf("can't create wfn from package: %v", err)
+		}
+
+		node.CPEMatch = append(node.CPEMatch, &nvd.NVDCVEFeedJSON10DefCPEMatch{
+			Cpe22Uri:   pkgAttrs.BindToURI(),
+			Cpe23Uri:   pkgAttrs.BindToFmtString(),
+			Vulnerable: false,
+		})
+	}
+
+	return &node, nil
+}
+
+func (ps *PackageState) createNode() (*nvd.NVDCVEFeedJSON10DefNode, error) {
+	pkgAttrs, err := packageName2wfn(ps.PackageName)
+	if err != nil {
+		return nil, fmt.Errorf("can't create wfn from package name: %v", err)
+	}
+
+	node := nvd.NVDCVEFeedJSON10DefNode{
+		Operator: "AND",
+		CPEMatch: []*nvd.NVDCVEFeedJSON10DefCPEMatch{
+			// package
+			{
+				Cpe22Uri:   pkgAttrs.BindToURI(),
+				Cpe23Uri:   pkgAttrs.BindToFmtString(),
+				Vulnerable: IsVulnerable(ps.FixState),
```

Comment:
We've discussed that already, just for posterity: vulnerable identifies the vulnerable product in multi-product configuration and has nothing to do with matching logic. So I thing inferring it from package fix is incorrect. Instead it should be something like `if IsVulnerable() { PackageState.CPE } else { PackageState.CPE AND NOT PackageState.PackageName.CPE() } ` or something along the lines...

Reply:
Actually... I think it should be like it is.
The only thing I'd change is to set `vulnerable: false` for the distribution CPE.
The configuration should still be like that and say whether the product is vulnerable or not (which is determined by the fix state).
Because, when constructing configuration from affectedRelease, vulnerable is always false for the product CPE (because it's not vulnerable any more).
Having a NOT would cause false positive matching again, IMO. What do you think @skogtwin ?

Reply:
Hmm... I'm not sure about that, my understanding was that the advisory was reporting vulnerable product and package version/release where that vulnerability was fixed. But I might be wrong. Let's give it some real-world testing, we can always change it later.